tests/test_odilo.py

```diff
@@ -0,0 +1,342 @@
+        self.api.queue_response(200, content=hold_ok_json)
+
+        hold_info = self.api.place_hold(self.PATRON, self.PIN, self.licensepool, 'ejcepas@odilotid.es')
+        ok_(hold_info, msg="HoldInfo null --> place hold failed!")
+        print 'Hold ok: %s' % hold_info.identifier
+
+    #################
+    # Patron Activity tests
+    #################
+
+    def test_41_patron_activity_invalid_patron(self):
+        patron_not_found_data, patron_not_found_json = self.sample_json("error_patron_not_found.json")
+        self.api.queue_response(404, content=patron_not_found_json)
+
+        assert_raises(PatronNotFoundOnRemote, self.api.patron_activity, self.PATRON, self.PIN)
+
+        print 'Test patron activity --> invalid partron ok!'
+
+    def test_42_patron_activity(self):
+        print ('Patron activity test...')
+        patron_checkouts_data, patron_checkouts_json = self.sample_json("patron_checkouts.json")
+        patron_holds_data, patron_holds_json = self.sample_json("patron_holds.json")
+        self.api.queue_response(200, content=patron_checkouts_json)
+        self.api.queue_response(200, content=patron_holds_json)
+
+        loans_and_holds = self.api.patron_activity(self.PATRON, self.PIN)
+        if loans_and_holds:
+            print 'Found: %i loans and holds' % len(loans_and_holds)
+        else:
+            print 'No loans or holds found'
+
+    #################
+    # Checkin tests
+    #################
+
+    def test_51_checkin_patron_not_found(self):
+        patron_not_found_data, patron_not_found_json = self.sample_json("error_patron_not_found.json")
+        self.api.queue_response(404, content=patron_not_found_json)
+
+        assert_raises(PatronNotFoundOnRemote, self.api.checkin, self.PATRON, self.PIN, self.licensepool)
+
+        print 'Test checkin --> invalid patron ok!'
+
+    def test_52_checkin_checkout_not_found(self):
+        checkout_not_found_data, checkout_not_found_json = self.sample_json("error_checkout_not_found.json")
+        self.api.queue_response(404, content=checkout_not_found_json)
+
+        assert_raises(NotCheckedOut, self.api.checkin, self.PATRON, self.PIN, self.licensepool)
+
+        print 'Test checkin --> invalid checkout ok!'
+
+    def test_53_checkin(self):
+        print('Checkin test...')
+        checkout_data, checkout_json = self.sample_json("patron_checkouts.json")
+        self.api.queue_response(200, content=checkout_json)
+
+        checkin_data, checkin_json = self.sample_json("checkin_ok.json")
+        self.api.queue_response(200, content=checkin_json)
+
+        response = self.api.checkin(self.PATRON, self.PIN, self.licensepool)
+        eq_(response.status_code, 200,
+            msg="Response code != 200, cannot perform checkin for record: " + self.licensepool.identifier.identifier
+                + " patron: " + self.PATRON)
+
+        checkout_returned = response.json()
+        print 'Checkout returned: %s' % checkout_returned['id']
```

Comment:
If this is important enough to print, its (known) value should be asserted.

Reply:
ok, asserted!